### PR TITLE
metrics: refactor metrics, reduce code complexity

### PR DESCRIPTION
**Pull request by royguo**
1. Remove unnecessary abstraction, reduce complexity
2. Add extents to file mapping, help us find filename by extent id (useful for collaborative GC)

Signed-off-by: Kuankuan Guo <guokuankuan@bytedance.com>

### CODE REVIEW

**Comment by yhr** [comment]
CI seems to be broken at the moment :/

**Comment by metaspace** [comment]
> CI seems to be broken at the moment :/

Smoke test ran fine, the commit does not compile.

```text
In file included from plugin/zenfs/fs/metrics_sample.h:8,
                 from plugin/zenfs/fs/fs_zenfs.cc:22:
plugin/zenfs/fs/snapshot.h: In constructor 'rocksdb::ZoneExtentSnapshot::ZoneExtentSnapshot(const rocksdb::ZoneExtent&, std::string)':
plugin/zenfs/fs/snapshot.h:75:7: error: declaration of 'filename' shadows a member of 'rocksdb::ZoneExtentSnapshot' [-Werror=shadow]
   75 |       : start(extent.start_),
      |       ^
```

**Comment by royguo** [comment]
This is wired, compile runs correctly here in my gcc 10.0.2. Tried to rename the `filename` variable to `fname`...

**Comment by skyzh** [comment]
> This is wired, compile runs correctly here in my gcc 10.0.2. Tried to rename the `filename` variable to `fname`...

Did you rebase to latest main?

**Comment by metaspace** [comment]
> This is wired, compile runs correctly here in my gcc 10.0.2. Tried to rename the `filename` variable to `fname`...

It's a little silly that the build output is not reported. I will try to fix that early January.

File: fs/io_zenfs.h [hunk 1]
```diff
@@ -115,6 +115,7 @@ class ZoneFile {
   void EncodeSnapshotTo(std::string* output) { EncodeTo(output, 0); };
   void EncodeJson(std::ostream& json_stream);
   void MetadataSynced() { nr_synced_extents_ = extents_.size(); };
+  void MetadataUnsynced() { nr_synced_extents_ = 0; };
```
Comment (yhr) [inline fs/io_zenfs.h:118]: This is not being used anywhere and seem to be unrelated to the PR, please remove.

**Review by yhr — CHANGES_REQUESTED** [review]

File: fs/io_zenfs.h (hunk 1 above)
Comment (royguo) [reply fs/io_zenfs.h]: Fixed

**Review by yhr — APPROVED** [review]
